src/clients/AcrossConfigStoreClient/AcrossConfigStoreClient.ts

```diff
@@ -89,37 +105,43 @@ export class AcrossConfigStoreClient extends BaseAbstractClient {
     super(eventSearchConfig);
     this.firstBlockToSearch = eventSearchConfig.fromBlock;
     this.latestBlockSearched = 0;
-    this.rateModelDictionary = new across.rateModel.RateModelDictionary();
   }
 
-  // <-- START LEGACY CONFIGURATION OBJECTS -->
-  // @dev The following configuration objects are pre-UBA fee model configurations and are deprecated as of version
-  // 2 of the ConfigStore. They are kept here for backwards compatibility.
-
   getRateModelForBlockNumber(
     l1Token: string,
     originChainId: number | string,
     destinationChainId: number | string,
     blockNumber: number | undefined = undefined
-  ): across.constants.RateModel {
+  ): RateModel {
     // Use route-rate model if available, otherwise use default rate model for l1Token.
     const route = `${originChainId}-${destinationChainId}`;
     const routeRateModel = this.getRouteRateModelForBlockNumber(l1Token, route, blockNumber);
-    return routeRateModel ?? this.rateModelDictionary.getRateModelForBlockNumber(l1Token, blockNumber);
+    if (routeRateModel) {
+      return routeRateModel;
+    }
+
+    const defaultRateModelUpdate = sortEventsDescending(this.cumulativeRateModelUpdates).find(
```

Comment:
The cleanup here is to read rate model events directly from this class rather than rely on this client's `update()` method to store the events into a `RateModelDictionary` class and then read it from there.